src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/Methodologies/MethodologyImageServiceTests.cs

```diff
@@ -129,7 +130,101 @@ await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContex
         }
 
         [Fact]
-        public async Task Delete_InvalidFileType()
+        public async Task UnlinkAndDeleteIfOrphaned_FilesLinkedToOtherMethodologyVersions()
+        {
+            var methodology = new Methodology();
+            var anotherMethodology = new Methodology();
+
+            var imageFile1 = new MethodologyFile
+            {
+                Methodology = methodology,
+                File = new File
+                {
+                    RootPath = Guid.NewGuid(),
+                    Filename = "image1.png",
+                    Type = Image
+                }
+            };
+
+            var imageFile2 = new MethodologyFile
+            {
+                Methodology = methodology,
+                File = new File
+                {
+                    RootPath = Guid.NewGuid(),
+                    Filename = "image2.png",
+                    Type = Image
+                }
+            };
+            
+            var imageFile2UsedByAnotherMethodology = new MethodologyFile
+            {
+                Methodology = anotherMethodology,
+                File = imageFile2.File
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.Methodologies.AddAsync(methodology);
+                await contentDbContext.MethodologyFiles.AddRangeAsync(imageFile1, imageFile2,
+                    imageFile2UsedByAnotherMethodology);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var blobStorageService = new Mock<IBlobStorageService>(MockBehavior.Strict);
+
+            blobStorageService.Setup(mock =>
+                    mock.DeleteBlob(PrivateMethodologyFiles, imageFile1.Path()))
+                .Returns(Task.CompletedTask);
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var service = SetupMethodologyImageService(contentDbContext: contentDbContext,
+                    blobStorageService: blobStorageService.Object);
+
+                var result = await service.UnlinkAndDeleteIfOrphaned(methodology.Id, new List<Guid>
+                {
+                    imageFile1.File.Id,
+                    imageFile2.File.Id
+                });
+
+                Assert.True(result.IsRight);
```

Comment:
I think you can use result.AssertRight here from either extensions although you don't do anything with the result so not much difference. Also you could use AsList just above here.

Reply:
Yup good call, I think I just inherited this test code from another test.  Have updated these now across the board in this test class, and also added in an AssertLeft() - how did we not have an AssertLeft() before?!